Adding CDC information as a new page in the AuraDB docs (#323)
neo4j/docs-aura: modules/ROOT/pages/auradb/change-data-capture/overview.adoc

[[change-data-capture-overview]]
= Change Data Capture (CDC)
:description: This section covers changing transaction log enrichment mode using the Aura Console.

label:AuraDB-Professional[] 
label:AuraDB-Enterprise[]

> Change Data Capture tracks changes to your data by querying the transaction logs for Creates, Updates and Deletes, so that you can update other applications or data sources.

Author: https://neo4j.com/docs/cdc/current/ refers to "Change Data Capture"
https://neo4j.com/docs/cdc/current/getting-started/key-considerations/ uses "change data capture"

> Change Data Capture tracks changes to your data by querying the transaction logs for Creates, Updates and Deletes, so that you can update other applications or data sources.

Reviewer: No, I meant _Creates, Updates and Deletes_ . Also, there should be an Oxford comma in there.

For more information see link:{neo4j-docs-base-uri}/cdc/current[Neo4j DBMS -> Change Data Capture (CDC)].

Log enrichment mode can be set to `OFF`, `DIFF` or `FULL`. 
For more information see link:{neo4j-docs-base-uri}/cdc/current/getting-started/enrichment-mode/[Neo4j DBMS -> CDC -> Enrichment mode].

== Modifying log enrichment mode from within the Aura Console

View an instance’s current log enrichment mode by clicking on an instance in the Aura Console. 

Note that default log enrichment mode is `OFF` and change data capture is disabled until it is set as `DIFF` or `FULL`.

Only Admins can change the log enrichment mode. 

> If an Admin changes log enrichment mode from `DIFF` or `FULL` to `OFF` applications dependent on CDC will stop working. For more information see link:https://neo4j.com/docs/cdc/current/getting-started/enrichment-mode/#_disabling_log_enrichment[Neo4j DBMS -> CDC -> Disabling Log Enrichment].

Author: Let's drop "the" and make it even more concise

== Performing a restore in Aura with log enrichment mode enabled

> If you take a snapshot and restore it to the same instance, or create a new instance from that snapshot, by default log enrichment mode will be set to `OFF`. 
> If the original status of that instance had log enrichment mode set to `DIFF` or `FULL`, the status changes. 
> You need to proactively enable it.

Reviewer: ```suggestion
To record changes when you take a snapshot and restore it to the same instance, you need to proactively set the log enrichment mode to `DIFF` or `FULL` since it is set to `OFF` by default.
```
I'm a little unsure what you are trying to say here, but this is an attempt to make it a little clearer.

Author: @l-heemann could you please take a few screenshots of this flow for us to see how it works please?

Reviewer 2: I think screenshots need to wait until the button becomes available in Aura. Maybe somebody from the aura-console team who is working on this feature has access to a dev environment or mockup of what it will look like before we release it to customers. I doubt I'll have access to an aura environment with this button before it exists for customers.

For now I think we should maybe rephrase it to be more of a link to the CDC docs on this.

> CDC assumes that changes can be represented as a continuous stream of changes, each new change building on top of the state of the previous changes. 

Reviewer: ```suggestion
CDC assumes that changes can be represented as a continuous stream, each new change building on top of the state of the previous changes. 
```

Restoring a Neo4j database breaks this continuous stream. 
See the implications documented in link:{neo4j-docs-base-uri}/cdc/current[Neo4j DBMS -> CDC -> Backup and Restore Operations] before proceeding with restoring your backup.

== Pause and resume

This is currently unsupported for databases enabled with CDC.

== Design considerations for applications with CDC in Aura

> When designing applications that query using CDC's procedures, those applications track their position in the stream of changes using a cursor. 
> The cursor is valid for a limited time. For more information see link:{neo4j-docs-base-uri}/cdc/current[Neo4j DBMS -> CDC -> Transaction log retention].
> 
> Developers should therefore consider high availability for their application so that continuity is maintained in querying in the event the application fails. 
> For example, use Kubernetes to automatically restart the application or a 3rd party product like Apache Kafka or Confluent. 

Reviewer: ```suggestion
Applications that use CDC's procedures to query track their position in the stream of changes using a cursor.
This cursor is valid only for a limited time, see link:{neo4j-docs-base-uri}/cdc/current[Neo4j DBMS -> CDC -> Transaction log retention] for more information.
Therefore, developers should consider high availability for their applications to ensure that query continuity is maintained in the event of application failure.
For example, Kubernetes can be used to automatically restart the application, but also Apache Kafka or Confluent.
```
(Kubernetes is also a 3rd party)

[[change-data-capture-overview]]
= Change Data Capture (CDC)
:description: This section covers changing transaction log enrichment mode using the Aura Console.

> label:AuraDB-Professional[] 

Reviewer 2: CDC is not available in Aura Pro. There is a limited subset of customers who have Aura Enterprise databases, but use Aura Pro as a testing environment. For those, support may enable CDC on their Pro instances, if the customer asks for it.
